### DISCUSSION TITLE
[Bug] When restart oap got interruptedException

### DISCUSSION THREAD

**Discussion by crazychengmm** [apache/skywalking#12182] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking) 9.4

### What happened

restart OAP Service failed
![image](https://github.com/apache/skywalking/assets/23517542/0ddaf3f6-8bb0-4fe3-a29b-c130bec7429d)

### What you expected to happen

We reboot the OAP service, sometimes OAP would be start success, but 90% would get a interrupted exception.
![image](https://github.com/apache/skywalking/assets/23517542/b2d4477a-12ce-4f7d-8531-c38887b682ab)

Check the info log, it seems list pod, timeout with k8s api.
So i checked the skywalking source code, noticed that connect k8s using  io.fabric8.kubernetes.client,
found timeout can change.
----------
public static final String KUBERNETES_CONNECTION_TIMEOUT_SYSTEM_PROPERTY = "kubernetes.connection.timeout";
public static final String KUBERNETES_UPLOAD_REQUEST_TIMEOUT_SYSTEM_PROPERTY = "kubernetes.upload.request.timeout";
public static final String KUBERNETES_REQUEST_TIMEOUT_SYSTEM_PROPERTY = "kubernetes.request.timeout";
-----------
kubernetes.connection.timeout
kubernetes.request.timeout
-----------
Default value is 10000ms, try to set 15000ms, not work for me.

Here is my questions.
1. This issue is happed with restart the OAP
2. 2024-84-29 17:14:55, 599 io.kubernetes.client.informer.cache.ProcessorListener 96 ［pool-6-thread-1) ERROR 11 - [9.4.9-52005311 processor

using  io.fabric8.kubernetes.client  ... why came out error with io.kubernetes.client?
<img width="935" alt="image" src="https://github.com/apache/skywalking/assets/23517542/262a8d0d-46b1-49b8-9c6c-08eccce58e20">

### How to reproduce

how to slove the issue.

### Are you willing to submit a pull request to fix on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by crazychengmm** [comment]
Noticed that 9.4 using client with io.kubernetes.client. I was checking wrong source code which is 8.9.

**Comment by crazychengmm** [comment]
About timeout, OkhttpClient is used by io.kubernetes.client, which is set by 10s.
<img width="601" alt="image" src="https://github.com/apache/skywalking/assets/23517542/c73e746e-c3fd-4329-b8f9-76045b219ec7">

If k8s is big enough, api return over 10s, may get error with time out.

<img width="1340" alt="image" src="https://github.com/apache/skywalking/assets/23517542/35b03787-8724-4071-adff-3e99c30da343">

**Comment by lujiajing1126** [comment]
Simply add timeout may not help. What about using `limit`?

https://kubernetes.io/docs/reference/using-api/api-concepts/#retrieving-large-results-sets-in-chunks

**Comment by crazychengmm** [comment]
> Simply add timeout may not help. What about using `limit`?
> 
> https://kubernetes.io/docs/reference/using-api/api-concepts/#retrieving-large-results-sets-in-chunks

KubernetesPods.java


```
@SneakyThrows
    private KubernetesPods() {
        KubernetesClient.setDefault();

        final CoreV1Api coreV1Api = new CoreV1Api();
        final CacheBuilder<Object, Object> cacheBuilder =
            CacheBuilder.newBuilder()
                .expireAfterAccess(Duration.ofMinutes(5));

        podByIP = cacheBuilder.build(new CacheLoader<String, Optional<V1Pod>>() {
            @Override
            public Optional<V1Pod> load(String ip) throws Exception {
                return coreV1Api
                    .listPodForAllNamespaces(
                        null, null, String.format(FIELD_SELECTOR_PATTERN_POD_IP, ip),
                        null, null, null, null, null, null, null)
                    .getItems()
                    .stream()
                    .findFirst();
            }
        });

```
limit was not config, limit was set null.
If using limit should change the code.Is there any other idea, Thanks

**Reply by crazychengmm** [reply]
@wu-sheng @lujiajing1126  Big boss, 求翻牌

**Reply by wu-sheng** [reply]
I didn't face this, you could change codes to see whether it fixes.
